pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -91,6 +96,29 @@ func (l *projectLifecycle) sync(key string, orig *v3.Project) error {
 		}
 	}
 
+	// create cluster role for this cluster's templates and templateversions. cluster role here too since we want to grant access to templates resource
```

Comment:
Project catalog related calls might not be generic enough to be in the package. Any way we can define them in the template controllers? Given it's a part of sync method, there is no implications of creating a separate controller (would be a diff story if we used Lifecycle)